coreneuron/utils/memory_utils.cpp

```diff
@@ -70,12 +73,26 @@ double nrn_mallinfo(void) {
         KERN_SUCCESS)
         return (size_t)0L; /* Can't access? */
     return info.resident_size / (1024.0 * 1024.0);
-#elif defined HAVE_MALLOC_H
-    struct mallinfo m;
-    m = mallinfo();
-    mbs = (m.hblkhd + m.uordblks) / (1024.0 * 1024.0);
+#else
+    std::ifstream file;
+    std::stringstream buffer;
+    file.open("/proc/self/statm");
+    if (file.is_open()) {
+        buffer << file.rdbuf();
+        file.close();
+        unsigned long long int total_virtual_memory;
+        buffer >> total_virtual_memory;
+        mbs = (total_virtual_memory * sysconf(_SC_PAGESIZE)) / (1024.0 * 1024.0);
```

Comment:
They will for sure provide very different numbers:
On a Linux x86 workstation, RSS will provides what have been accessed, not what have been allocated.
On an HPC system with KNL: I have to check but it seems that we try to keep everything in Physical memory and the policy might be to reserve physical space directly. 
I will check if this behavior is correct.
Also I have checked that "data" section in statm is growing properly with allocation, so instead of full virtual memory, we can take this figure (without code and shared).